addons/website/models/website.py

```diff
@@ -1292,13 +1287,19 @@ def action_dashboard_redirect(self):
             return self.env["ir.actions.actions"]._for_xml_id("website.backend_dashboard")
         return self.env["ir.actions.actions"]._for_xml_id("website.action_website")
 
+    def get_client_action_url(self, url, mode_edit=False):
+        client_action_url = f'/web#action=website.website_preview&path={url}'
+        if mode_edit:
+            client_action_url += '&enable_editor=1'
+        return client_action_url
```

Comment:
here again, `{url}` shouldn't be inserted as text, because it can contains its own set of parameters that would be lost and mixed with those of the fragment params.

Why not use `url_encode` to format all the parameters?

```suggestion
    def get_client_action_url(self, url, mode_edit=False):
        action_params = {
            "action": "website.website_preview",
            "path": url,
        }
        if mode_edit:
            action_params["enable_editor"] = 1
        return "/web#" + url_encode(action_params)
```